app/models/account.rb

```diff
@@ -1,6 +1,7 @@
 class Account < ApplicationRecord
   enum status: {active: 0, ban: 1}
-  attr_accessor :remember_id
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```

Comment:
rememberable work flow của nó khác gì so với tutorial em nhỉ
Anh muốn thay đổi độ phức tạp của mật khẩu thì cần phải sửa ở đâu

Reply:
1.Rememberable work flow
Dạ theo như em tìm hiểu ở [đây](https://security.stackexchange.com/questions/182292/ruby-devise-salt-exposed) là nếu người dùng chọn tùy chọn "Remember me" khi đăng nhập, Devise tạo một cookie (remember_user_token) chứa thông tin để duy trì trạng thái đăng nhập của người dùng có dạng:
 [[<user_id>], "<user_salt>", "<cookie_generated_at_UTC_time>"]]
Khi ứng dụng nhận lại cookie đã ký từ trình duyệt, nó sẽ sử dụng thông tin trong cookie để xác thực lại user bằng cách : Phân tích cookie để lấy user_id, user_salt, và thời gian tạo cookie, tìm người dùng trong cơ sở dữ liệu với user_id tương ứng, so sánh user_salt từ cookie với user_salt của người dùng trong cơ sở dữ liệu.Nếu user_salt khớp, người dùng được xác thực và phiên đăng nhập được thiết lập. Nếu không khớp hoặc cookie đã hết hạn, người dùng sẽ không được xác thực và phải đăng nhập lại.
-> Từ đó ko cần phải lưu remember_digest trong db
2. Thay đổi độ phức tạp mật khẩu 
Sử dụng validate và tạo hàm password_complexity
validate :password_complexity

def password_complexity
  if password.present?
     if !password.match(/^(?=.*[a-z])(?=.*[A-Z])/) 
       errors.add :password, "Password complexity requirement not met"
     end
  end
end